runCmdStanTests.py

```diff
@@ -144,12 +145,18 @@ def main():
     if len(sys.argv) < 2:
         usage()
 
+    stan_mpi = False
     try:
         with open('make/local') as f:
-            stan_mpi =  'STAN_MPI' in f.read()
+            for (_, line) in enumerate(f):
+                if line.strip().startswith("#"):
+                    continue
+                else:
+                    stan_mpi = line.find('STAN_MPI=true') != -1
```

Comment:
[not sure if it's important] This won't accurately detect things like `STAN_MPI=1`, which also enable MPI

Reply:
It wasn't detecting that before. I think if we want to do that I'd rather do it in a different PR

Reply:
It was searching for STAN_MPI, so it would have. you've just traded false positives for false negatives here

Reply:
Oh I thought you meant an environment variable the user set. I can open another PR to just make this search for `STAN_MPI`